cni/pkg/ambient/informers.go

```diff
@@ -85,13 +86,25 @@ func (s *Server) EnqueueNamespace(o controllers.Object) {
 
 func (s *Server) Reconcile(input any) error {
 	event := input.(controllers.Event)
-	log := log.WithLabels("type", event.Event)
 	pod := event.Latest().(*corev1.Pod)
+	log := log.WithLabels("type", event.Event, "pod", config.NamespacedName(pod))
 	if ztunnelPod(pod) {
+		log.Debugf("reconciling ztunnel")
 		return s.ReconcileZtunnel()
 	}
+	log.Debugf("reconciling pod")
 	switch event.Event {
 	case controllers.EventAdd:
+		ns := s.namespaces.Get(pod.Namespace, "")
+		if ns == nil {
+			return fmt.Errorf("failed to find namespace %v", ns)
+		}
+		// Typically, a pod Add is handled by the CNI plugin.
+		// But if CNI restarts, we clear the rules, so this can happen due to CNI restart as well
+		if ambientpod.PodZtunnelEnabled(ns, pod) && !IsPodInIpset(pod) {
```

Comment:
```suggestion
		if ambientpod.PodZtunnelEnabled(ns, pod) && !IsPodInIpset(pod) {
```

The issue here is that the pod is missing from the ipset, right?

IIRC the ipset is only used for DNS redirection, correct?

Reply:
Its missing all the rules, and I think ipset is used for more than DNS (or it used to be). But its broken without DNS at least, regardless of my explanation's accuracy

Reply:
IsPodInIpset was just a signal of "Does pod have redirection rules"; maybe we need a more accurate measure of that.

Reply:
There isn't really an accurate measure of that - and anyway I think we need to repair-and-reconcile, informer doesn't need to care about the status of redirection, it should just call `Add/Delete` - the former even for updates.